sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java

```diff
@@ -454,9 +455,20 @@ public void deleteById(String containerName, Object id, PartitionKey partitionKe
         Assert.notNull(ids, "Id list should not be null");
         Assert.notNull(domainType, "domainType should not be null.");
         Assert.hasText(containerName, "container should not be null, empty or only whitespaces");
-
+        final List<Object> idList = new ArrayList<>();
```

Comment:
I think we need this change for `ReactiveCosmosTemplate` as well.

Reply:
Should be, but there's o similar method in ReactiveCosmosTemplate,  I unified the procesing of ID type conversion.

Reply:
Good, makes sense.